pkg/arguments/arguments.go

```diff
@@ -185,6 +187,91 @@ func ParseKnownFlags(cmd *cobra.Command, argv []string, failOnUnknown bool) erro
 	return nil
 }
 
+// PreprocessUnknownFlagsWithId Parses known and unknown flags will take the command line arguments and map the ones
+// that fit with known flags.
+func PreprocessUnknownFlagsWithId(cmd *cobra.Command, argv []string) error {
+	flags := cmd.Flags()
+
+	var validArgs []string
+	var upcomingValue bool
+
+	foundId := false
+	for _, arg := range argv {
+		switch {
+		// Upcoming value from a space-separated value
+		case upcomingValue:
+			if strings.HasPrefix(arg, "-") {
+				return fmt.Errorf("No value given for flag '%s'", arg)
+			}
+			validArgs = append(validArgs, arg)
+			upcomingValue = false
+		// A long flag with a space separated value
+		case strings.HasPrefix(arg, "--") && !strings.Contains(arg, "="):
+			flagName := arg[2:]
+			// Skip EOF and known flags
+			if flag := flags.Lookup(flagName); flag != nil {
+				validArgs = append(validArgs, arg)
+				if flag.Value.Type() != boolType {
+					upcomingValue = true
+				}
+			} else {
+				upcomingValue = true
+			}
+		// A long flag with a value after an equal sign
+		case strings.HasPrefix(arg, "--") && strings.Contains(arg, "="):
+			flagName := strings.SplitN(arg[2:], "=", 2)[0]
+			if flags.Lookup(flagName) != nil {
+				validArgs = append(validArgs, arg)
+			}
+			upcomingValue = false
+		// A short flag with a space separated value
+		case strings.HasPrefix(arg, "-") && !strings.Contains(arg, "="):
+			flagName := arg[1:]
+			if flag := flags.Lookup(flagName); flag != nil {
+				validArgs = append(validArgs, arg)
+				if flag.Value.Type() != boolType {
+					upcomingValue = true
+				}
+			} else {
+				upcomingValue = true
+			}
+		// A short flag with with a value after an equal sign
+		case strings.HasPrefix(arg, "-") && strings.Contains(arg, "="):
+			flagName := strings.SplitN(arg[1:], "=", 2)[0]
+			if flags.Lookup(flagName) != nil {
+				validArgs = append(validArgs, arg)
+			}
+			upcomingValue = false
+		default:
+			foundId = true
+		}
+	}
+
+	err := flags.Parse(validArgs)
+	if err != nil {
+		return err
+	}
+
+	// If help is called, regardless of other flags, return we want help.
+	// Also say we need help if the command isn't runnable.
+	helpVal, err := cmd.Flags().GetBool("help")
+	if err != nil {
+		// should be impossible to get here as we always declare a help
+		// flag in InitDefaultHelpFlag()
+		cmd.Println("\"help\" flag declared as non-bool. Please correct your code")
```

Comment:
@hunterkepley If we don't expect this to happen and cannot recover from it, shouldn't we `panic`? I feel that's much more likely to get peoples attention rather than a log message.